### PR TITLE
Add identity specific tests

### PR DESCRIPTION
**Pull request by VladimirWork**

### CODE REVIEW

File: tests/e2e/test_identity.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+from vdrtools import cheqd_keys, cheqd_pool, cheqd_ledger
+from vdrtools.error import CommonInvalidStructure
+
+from helpers import create_did_helper, query_did_helper, random_string, update_did_helper, wallet_helper, get_base_account_number_and_sequence, get_timeout_height, get_balance_vdr, send_tx_helper, \
+    SENDER_ADDRESS, SENDER_MNEMONIC, RECEIVER_ADDRESS, LOCAL_NET_NETWORK, TEST_NET_GAS_X_GAS_PRICES_INT, GAS_AMOUNT, GAS_PRICE, DENOM
+
+# logger = logging.getLogger(__name__)
+# logging.basicConfig(level=logging.DEBUG)
+
+key_alias = "operator0"
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py:18]: I think it should be constants with upper letters as below ones.

File: tests/e2e/test_identity.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        await cheqd_keys.add_from_mnemonic(wallet_handle, key_alias, SENDER_MNEMONIC, "")
+    )["pub_key"]
+
+    msg = await cheqd_ledger.bank.build_msg_send(
+        SENDER_ADDRESS, RECEIVER_ADDRESS, transfer_amount, DENOM
+    )
+    account_number, sequence_number = await get_base_account_number_and_sequence(pool_alias, SENDER_ADDRESS)
+    timeout_height = await get_timeout_height(pool_alias)
+    test_tx = await cheqd_ledger.auth.build_tx(
+        pool_alias, public_key, msg, account_number, sequence_number, 0, 0, DENOM, SENDER_ADDRESS, timeout_height, default_memo
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py:51]: I think this zeros also should constants.

File: tests/e2e/test_identity.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    )
+    request = await cheqd_ledger.tx.build_query_simulate(test_tx)
+    response = await cheqd_pool.abci_query(pool_alias, request)
+    response = await cheqd_ledger.tx.parse_query_simulate_resp(response)
+    gas_estimation = json.loads(response)["gas_info"]["gas_used"]
+
+    # negative case
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py:58]: So, maybe it should be divided into different tests? like `test_some_positive` and `test_some_negative`.
Cause mixing several cases in the one test can be not clear.

File: tests/e2e/test_identity.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    wallet_handle, _, _ = await wallet_helper()
+    public_key = json.loads(
+        await cheqd_keys.add_from_mnemonic(wallet_handle, key_alias, SENDER_MNEMONIC, "")
+    )["pub_key"]
+
+    msg = await cheqd_ledger.bank.build_msg_send(
+        SENDER_ADDRESS, RECEIVER_ADDRESS, str(default_amount), DENOM
+    )
+    _, tx_hash = await send_tx_helper(pool_alias, wallet_handle, key_alias, public_key, SENDER_ADDRESS, msg, memo) 
+    # time.sleep(15) # FIXME
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py]: Is it still needed?

File: tests/e2e/test_identity.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        await cheqd_pool.broadcast_tx_commit(pool_alias, prod_tx_negative_signed)
+
+    # positive case
+    account_number, sequence_number = await get_base_account_number_and_sequence(pool_alias, SENDER_ADDRESS) # get this one more time to avoid `incorrect account sequence` error
+    prod_tx = await cheqd_ledger.auth.build_tx(
+        pool_alias, public_key, msg, account_number, sequence_number, int(gas_estimation*magic_number_positive), int(gas_estimation*magic_number_positive*GAS_PRICE), DENOM, SENDER_ADDRESS, timeout_height, default_memo
+    )
+    prod_tx_signed = await cheqd_ledger.auth.sign_tx(wallet_handle, key_alias, prod_tx)
+    positive_res = await cheqd_pool.broadcast_tx_commit(pool_alias, prod_tx_signed)
+    assert json.loads(positive_res)["check_tx"]["code"] == 0
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py:73]: Here is magic_number also.

File: tests/e2e/test_identity.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+
+@pytest.mark.parametrize("transfer_amount", [1, 999, 1001, 987654321])
+@pytest.mark.asyncio
+async def test_token_transfer(transfer_amount):
```
Comment (anikitinDSR) [inline tests/e2e/test_identity.py:78]: Do we have test case for transferring tokens more than we have?

Comment (VladimirWork) [reply tests/e2e/test_identity.py]: Added.

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**